### PR TITLE
chore: use Red Hat images

### PR DESCRIPTION
**Pull request by matejvasek**
- Use Red Hat productized images.
- Skip some test since images are not published yet.

### CODE REVIEW

File: openshift/patches/0006-patch-images.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+diff --git a/k8s/dialer.go b/k8s/dialer.go
+index 306e5b03..83027db1 100644
+--- a/k8s/dialer.go
++++ b/k8s/dialer.go
+@@ -24,7 +24,7 @@ import (
+ )
+ 
+ const (
+-	socatImage = "quay.io/boson/alpine-socat:1.7.4.3-r1-non-root"
++	socatImage = "registry.redhat.io/openshift-serverless-1/func"
```
Comment (Kaustubh-pande) [inline openshift/patches/0006-patch-images.patch:10]: The name should be `registry.redhat.io/openshift-serverless-1/kn-plugin-func-rhel8`

Comment (lance) [inline openshift/patches/0006-patch-images.patch:10]: From @Kaustubh-pande the name is [registry.redhat.io/openshift-serverless-1/kn-plugin-func-rhel8](http://registry.redhat.io/openshift-serverless-1/kn-plugin-func-rhel8)

**Comment by matejvasek** [comment]
@jrangelramos PTAL

**Comment by jrangelramos** [comment]
For the on cluster test piece it looks good to me.

**Comment by matejvasek** [comment]
@lance @dsimansk PTAL

**Comment by dsimansk** [comment]
@matejvasek to run the tests & verify it. You'll need to apply the patch to `release-next` branch and create a new PR to run those.

**Comment by matejvasek** [comment]
@dsimansk I verified that it works in upstream.

**Comment by matejvasek** [comment]
https://github.com/knative-sandbox/kn-plugin-func/pull/1251

**Pull request merged**